gui/wxpython/mapdisp/settings.py

```diff
@@ -0,0 +1,276 @@
+        )
+        ChBItem.__init__(self, parent, mapWindowProperties, label, tooltip)
+
+    @property
+    def mapWindowProperty(self):
+        return self._properties.alignExtent
+
+    @mapWindowProperty.setter
+    def mapWindowProperty(self, value):
+        self._properties.alignExtent = value
+
+    def mapWindowPropertyChanged(self):
+        return self._properties.alignExtentChanged
+
+
+class ChBResolution(ChBItem):
+    """Checkbox to select used display resolution."""
+
+    def __init__(
+        self,
+        parent,
+        giface,
+        mapWindowProperties,
+        label=_("Constrain display resolution to computational settings"),
+    ):
+        self.giface = giface
+        tooltip = _(
+            "Constrain display resolution "
+            "to computational region settings. "
+            "Default value for new map displays can "
+            "be set up in 'User GUI settings' dialog."
+        )
+        ChBItem.__init__(self, parent, mapWindowProperties, label, tooltip)
+
+    @property
+    def mapWindowProperty(self):
+        return self._properties.resolution
+
+    @mapWindowProperty.setter
+    def mapWindowProperty(self, value):
+        self._properties.resolution = value
+
+    def mapWindowPropertyChanged(self):
+        return self._properties.resolutionChanged
+
+    def _onToggleCheckBox(self, event):
+        """Update display when toggle display mode"""
+        super()._onToggleCheckBox(event)
+
+        # redraw map if auto-rendering is enabled
+        if self._properties.autoRender:
+            self.giface.updateMap.emit()
+
+
+class ChBShowRegion(ChBItem):
+    """Checkbox to enable and disable showing of computational region."""
+
+    def __init__(
+        self, parent, giface, mapWindowProperties, label=_("Show computational extent")
+    ):
+        self.giface = giface
+        tooltip = _(
+            "Show/hide computational "
+            "region extent (set with g.region). "
+            "Display region drawn as a blue box inside the "
+            "computational region, "
+            "computational region inside a display region "
+            "as a red box)."
+        )
+        ChBItem.__init__(self, parent, mapWindowProperties, label, tooltip)
+
+    @property
+    def mapWindowProperty(self):
+        return self._properties.showRegion
+
+    @mapWindowProperty.setter
+    def mapWindowProperty(self, value):
+        self._properties.showRegion = value
+
+    def mapWindowPropertyChanged(self):
+        return self._properties.showRegionChanged
+
+    def _onToggleCheckBox(self, event):
+        """Shows/Hides extent (comp. region) in map canvas.
+
+        Shows or hides according to checkbox value.
+        """
+        super()._onToggleCheckBox(event)
+
+        # redraw map if auto-rendering is enabled
+        if self._properties.autoRender:
+            self.giface.updateMap.emit(render=False)
+
+
+class MapDisplayPreferencesDialog(PreferencesBaseDialog):

```

Comment:
I am not sure we should be using PreferencesBaseDialog at all, we are not setting GUI settings here.

Reply:
I was thinking that these settings could replace the same settings in Map Display. If you test this PR it looks very intuitive, you just set up something and if you want, you can save it and use it later on.

Reply:
I am still thinking about it. You are probably right that It should influence only the given mapdisplay or statusbar. But then I am not sure about the Projection page. We agreed to move it to Map Display settings, but it is also in the form of global settings. So generally speaking, do we want to get rid of the global settings a little bit? Or do we just want to add some functions from GUI settings in the form of Local settings to Map Display settings?...

Reply:
You could potentially set the defaults from there, but I think it would be confusing and more work. Regarding the projection, I would remove it from general settings altogether. I would keep the rest there. Ultimately we are trying to reduce the amount of work.